### DISCUSSION TITLE
Anyone using the graphqlite bundle with multiple namespaces for types and controllers

### DISCUSSION THREAD

**Discussion by aszenz** [thecodingmachine/graphqlite#385] [General]
I recently added a new namespace for types and controllers to the bundle config but got some weird errors related to not finding any fields for graphql types. Looks like the bundle was searching for types in the wrong directory. When i moved the code into the same namespace it all worked again, so I'm just wondering if there's a bug with the types finding logic when configured with multiple namespaces

**Comment by aszenz** [comment]
I received these kinds of errors:

`The GraphQL object type \"Propagation\" has no fields defined. Please check that some fields are defined (using the @Field annotation). If some fields are defined, please check that at least one is visible to the current user.",`

The propagation object type was defined with fields, but it seems to be looking at the actual class not the external type class

**Comment by oojacoboo** [comment]
It's hard to say for sure without more details.  Have you tried clearing your cache, whichever you've specified in your `SchemaFactory`?

**Reply by aszenz** [reply]
yeah two devs reproduced it, so cache doesn't seem to be the issue. i'll need to setup a minimal reproducible example to provide more details, but i'm just looking to confirm that people are using multiple namespaces without issues right

**Reply by oojacoboo** [reply]
We use multiple defined namespaces within our `SchemaFactory` config:

```php
$factory->addTypeNamespace('Foo\\Bar');
$factory->addTypeNamespace('Foo\\Baz');
```

Is this what you're referring to here?

**Comment by aszenz** [comment]
I mean like this, we use php config `graphqlite.php` for symfony:

```php
<?php

declare(strict_types=1);

namespace Symfony\Component\DependencyInjection\Loader\Configurator;

use Symfony\Config\GraphqliteConfig;

return static function (ContainerConfigurator $containerConfigurator, GraphqliteConfig $config): void {
    $config
        ->namespace()
        ->controllers([
            'Corp\Order\UI\GraphQL\Resolvers\\',
            'Corp\Foo\Bar\Z\Resolvers\\',
        ])
        ->types([
            'Corp\Order\UI\GraphQL\Types\\',
            'Corp\Foo\Bar\Z\Types\\',
        ])
    ;
};
```

**Reply by oojacoboo** [reply]
Sorry, I can't help you with the Symfony bundle.  Really, those questions should be directed towards the symfony bundle lib.  I'd assume that those methods, `controllers` and `types` correspond to the `addControllerNamespace` and `addTypeNamespace` respectively.  But, I really don't know - don't use it.

**Reply by aszenz** [reply]
u r right, it should be reported in the symfony bundle repo, the code does use these methods 

```php
   foreach ($controllersNamespaces as $controllersNamespace) {
            $schemaFactory->addMethodCall('addControllerNamespace', [ $controllersNamespace ]);
            foreach ($this->getClassList($controllersNamespace) as $className => $refClass) {
                $this->makePublicInjectedServices($refClass, $reader, $container, true);
            }
        }

        foreach ($typesNamespaces as $typeNamespace) {
            $schemaFactory->addMethodCall('addTypeNamespace', [ $typeNamespace ]);
            foreach ($this->getClassList($typeNamespace) as $className => $refClass) {
                $this->makePublicInjectedServices($refClass, $reader, $container, false);
            }
        }
```
i'll move this discussion there

**Comment by aszenz** [comment]
Created issue here

https://github.com/thecodingmachine/graphqlite-bundle/issues/103
